frontend/src/pages/Tasks.tsx

```diff
@@ -2,26 +2,33 @@ import { useEffect, useMemo, useState } from "react";
 import { useTasks } from "../api/backendComponents";
 import Layout from "./layout";
 import { CtfTaskReadModel } from "../api/backendSchemas";
-import { useFirstBloodNotification } from "../hooks/useFirstBloodNotification";
-import { Bounce, ToastContainer, toast } from "react-toastify";
 import "react-toastify/dist/ReactToastify.css";
 import TaskGroupComponent from "../components/tasks/taskGroupComponent";
 import Toggle from "../components/toggle";
 import style from "./tasks.module.css";
 import GroupList from "../components/tasks/groupList";
-import classNames from "classnames";
+import { useTaskRefresher } from "../hooks/useTaskRefresh";
 
 export type GroupedTasks = Record<string, CtfTaskReadModel[]>;
 
 export default function Tasks() {
-  const { data, isLoading, error } = useTasks({});
-  const firstBloodNotification = useFirstBloodNotification();
+  const { data, isLoading, error, refetch } = useTasks({});
+  const { refresh, setRefresh } = useTaskRefresher();
 
   const [showSolvedTasks, setShowSolvedTasks] = useState(true);
   const [currentGroup, setCurrentGroup] = useState<keyof GroupedTasks | null>(
     null,
   );
 
+  //Refetch on websocket signal
+  useEffect(() => {
+    const refetchAndSet = async () => {
+      await refetch();
+      setRefresh(false);
+    };
+    refetchAndSet();
+  }, [refresh]);
+
```

Comment:
If you rename `refetch` to `lastNewTaskDate` and populate it with `new Date()` then you do not need to run `setRefresh(false)`

Reply:
Do you mean so that it verifies that the last Date() was over X minutes ago? Since the frontend doesn't really care for Date, as it just gets a signal to refetch

Reply:
If you use Date as signal instead of boolean, then the useEffect will trigger on every new change and you do not need to manually put the boolean back to false to trigger it the next time.

Reply:
Smart! It should be good now! Me and daniel just relaised we over-engineered the Toast thing, so perhaps wait until late tonight for me to remove that :)